### PR TITLE
[SPARK-31298][SQL] Create DataSource External Table path also need non-existent or empty

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
When create MANAGED table, we always use default location, when validate the tableLocation is wrong?

When create External Catalog, we also need to check table location to avoid rewrite data existed in location.

### Why are the changes needed?
Avoid overwrite other data we needed 

### Does this PR introduce any user-facing change?
When create external table,  also need to check location

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
@dongjoon-hyun @wangyum 
Maybe we should check this part first then do same change in Hive part, seems my pr https://github.com/apache/spark/pull/25290 's code is complex, here we can reuse some code.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala [hunk 1]
```diff
@@ -328,15 +328,17 @@ class SessionCatalog(
...
-    if (table.tableType == CatalogTableType.MANAGED) {
-      val tableLocation =
-        new Path(table.storage.locationUri.getOrElse(defaultTablePath(table.identifier)))
-      val fs = tableLocation.getFileSystem(hadoopConf)
-
-      if (fs.exists(tableLocation) && fs.listStatus(tableLocation).nonEmpty) {
-        throw new AnalysisException(s"Can not create the managed table('${table.identifier}')" +
-          s". The associated location('${tableLocation.toString}') already exists.")
-      }
+    // SPARK-31298: when create external table also should be a non-existent or empty
```
Comment (dongjoon-hyun) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala]: This looks wrong, @AngersZhuuuu . `EXTERNAL TABLE` is used to mount an exiting table location.

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]

**Comment by dongjoon-hyun** [comment]
Please see the definition.
- https://cwiki.apache.org/confluence/display/Hive/Managed+vs.+External+Tables

I'll close this PR first. We can discussion more at this PR.

**Comment by dongjoon-hyun** [comment]
Although this PR is wrong, you may refocus SPARK-31298 on what is the problem you have. So, I still keep SPARK-31298 open.

**Pull request closed without merging**